disnake/file.py

```diff
@@ -111,3 +112,29 @@ def close(self) -> None:
         self.fp.close = self._closer
         if self._owner:
             self._closer()
+
+    def __sizeof__(self) -> int:
+        return sys.getsizeof(self.fp)
```

Comment:
What's the need for `__sizeof__`?

Reply:
Since the property `size` is being suggested to be implemented, it would be appropriate to implement `__sizeof__` and make the property depend on the dunder since dunders are defined to execute special or "magic" tasks hence the synonym "magic methods" it would be more appropriate to let the method carry on with its duty, it's also better to implement it as if a user uses `sys.getsizeof` on the `File` object it would work appropriately and wont depend on a property if the method gets implemented in such pattern.

Reply:
That doesn't make sense, `sys.getsizeof`/`__sizeof__` is meant for the size of the object in memory, not the size of the file on disk.
The new `bytes_length` property added here should be sufficient.

Reply:
```py
In [1]: from pathlib import Path

In [2]: import os, sys

In [3]: Path('test.dat').write_bytes(b'a' * 1_000_000)
Out[3]: 1000000

In [4]: fp = Path('test.dat').open('rb')

In [5]: sys.getsizeof(fp)
Out[5]: 4264

In [6]: os.stat(fp.fileno()).st_size
Out[6]: 1000000
```